### ISSUE TITLE
Persist port-forward configuration in localconfig

### ISSUE THREAD

**Issue by scjudd** [argoproj/argo-cd#7226]
# Summary

We should allow `port-forward-namespace` to be specified in the `servers` entries in `~/.argocd/config`.

# Motivation

I have multiple Argo CD servers in different namespaces. I currently connect to these servers by first switching to the correct context via `argocd context foo` and then pass the `--port-forward-namespace foo` parameter to all CLI commands.

It would make for a better experience if the port-forward-namespace could be remembered as part of the current context, much in the same way that kubectl associates a namespace with a context.

# Proposal

We should add `PortForwardNamespace` to `util/localconfig/localconfig.go` and update the code paths which respect `--port-forward-namespace` to first consider the value from the current context.

**Comment by wanghong230** [comment]
Any reason to not give a load balancer to your argocd(s)?
port-forward-namespace and port-forward are used for troubleshooting and debugging in most the senerios.

**Comment by scjudd** [comment]
Yeah the LoadBalancer works fine, and maybe this use case is a bit contrived, I just thought that the port-forward feature could be a little less cumbersome if it could be used with the CLI contexts.

I had the a colleague make the same suggestion (i.e., just don't use port-forwarding) out-of-band, but it's not due to a lack of a workable solution that I've raised this. This is a proposal on how we might make the feature feel like less of an afterthought. If contexts are meant to hold the connection and authentication information for an ArgoCD installation, and one apparently supported method of connecting and authenticating is via port-forwarding, it just logically follows that port-forward configuration should be able to be persisted to the context, IMO.

It seems _weird_ that we would write a 'port-forward' section to `~/.argocd/config` after using the port-forward command flags once, but not actually persist anything useful regarding port-forwarding to the config. 🤷‍♂️
